src/main/java/com/mraof/minestuck/item/StoneTabletItem.java

```diff
@@ -0,0 +1,59 @@
+package com.mraof.minestuck.item;
+
+import com.mraof.minestuck.Minestuck;
+import com.mraof.minestuck.client.gui.StoneTabletScreen;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.util.ITooltipFlag;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.ActionResultType;
+import net.minecraft.util.Hand;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraft.world.World;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+public class StoneTabletItem extends Item
+{
+	public StoneTabletItem(Properties properties)
+	{
+		super(properties);
+		addPropertyOverride(new ResourceLocation(Minestuck.MOD_ID, "carved"), (stack, world, holder) -> hasText(stack) ? 1 : 0);
+	}
+	
+	@Override
+	public void addInformation(ItemStack stack, @Nullable World worldIn, List<ITextComponent> tooltip, ITooltipFlag flagIn)
+	{
+		super.addInformation(stack, worldIn, tooltip, flagIn);
+		
+		if(hasText(stack))
+			tooltip.add(new TranslationTextComponent(getTranslationKey()+".carved").applyTextStyle(TextFormatting.GRAY));
+	}
+	
+	@Override
+	public ActionResult<ItemStack> onItemRightClick(World worldIn, PlayerEntity playerIn, Hand handIn)
+	{
+		ItemStack stack = playerIn.getHeldItem(handIn);
+		if(worldIn.isRemote)
+		{
+			boolean canEdit = playerIn.getHeldItem(handIn == Hand.MAIN_HAND ? Hand.OFF_HAND : Hand.MAIN_HAND).isItemEqual(new ItemStack(MSItems.CARVING_TOOL));
+			String text = hasText(stack) ? stack.getTag().getString("text") : "";
+			Minecraft.getInstance().displayGuiScreen(new StoneTabletScreen(playerIn, handIn, text, canEdit));
```

Comment:
This might cause problems with dedicated servers since you refer to a screen class and the minecraft class here. Have you tested this with dedicated servers? If it does cause a problem, consider moving the gui display to MSScreenFactories.